ansible-tests/validations/library/validate_network_environment.py

```diff
@@ -0,0 +1,170 @@
+        if item.endswith('NetCidr'):
+            cidrinfo[item] = data
+        elif item.endswith('AllocationPools'):
+            poolsinfo[item] = data
+        elif item.endswith('NetworkVlanID'):
+            vlaninfo[item] = data
+        elif item == 'ExternalInterfaceDefaultRoute':
+            routeinfo = data
+        elif item == 'BondInterfaceOvsOptions':
+            bondinfo = data
+
+    errors.extend(check_cidr_overlap(cidrinfo.values()))
+    errors.extend(
+        check_allocation_pools_pairing(
+            network_data.get('parameter_defaults', {}), poolsinfo))
+    errors.extend(check_vlan_ids(vlaninfo))
+
+    return errors
+
+
+def validate_nic_configs(resource, path):
+    errors = []
+    nic_data = {}
+    try:
+        with open(path, 'r') as nic_file:
+            nic_data = yaml.load(nic_file)
+    except IOError:
+        errors.append('The resource "{}" reference file does not exist: "{}"'
+                      .format(resource, path))
+
+    # Look though every resources bridges and make sure there is only a single
+    # bond per bridge and only 1 interface per bridge if there are no bonds.
+    for name, resource in six.iteritems(nic_data.get('resources', {})):
+        bridges = resource['properties']['config']['os_net_config']['network_config']
+        for bridge in bridges:
+            if bridge['type'] == 'ovs_bridge':
+                bond_count = 0
+                interface_count = 0
+                for bond in bridge['members']:
+                    if bond['type'] == 'ovs_bond':
+                        bond_count += 1
+                    if bond['type'] == 'interface':
+                        interface_count += 1
+                if bond_count == 2:
+                    errors.append(
+                        'Invalid bonding: There are 2 bonds for '
+                        ' bridge {} of resource {} in {}'.format(
+                            bridge['name'], name, path))
+                if bond_count == 0 and interface_count > 1:
+                    errors.append(
+                        'Invalid interface: When not using a bond, '
+                        'there can only be 1 interface for bridge {} '
+                        'of resource {} in {}'.format(
+                            bridge['name'], name, path))
+    return errors
+
+
+def check_cidr_overlap(networks):
+    errors = []
+    objs = []
+    for x in networks:
+        try:
+            objs += [netaddr.IPNetwork(x.decode('utf-8'))]
+        except ValueError:
+            errors.append('Invalid address: {}'.format(x))
+
+    for net1, net2 in itertools.combinations(objs, 2):
+        if (net1 in net2 or net2 in net1):
+            errors.append(
+                'Overlapping networks detected {} {}'.format(net1, net2))
+    return errors
+
+
+def check_allocation_pools_pairing(filedata, pools):
+    errors = []
+    for poolitem in pools:
+        pooldata = filedata[poolitem]
+        pool_objs = [netaddr.IPRange(
```

Comment:
I'll do that here because it's easy enough, but note that this patch just takes the existing [network-environment-validator code](https://github.com/rthallisey/clapper/blob/7d0afab6b62c36317c347ccf959b6cdc409b1040/network-environment-validator.py) and makes it runnable through Ansible, too.

Plus, the original code is littered with things like this -- there's a ton more we can look for. So my intention with this PR is to just port it over and then keep improving on it in subsequent patches.